src/lib/support/ZclString.h

```diff
@@ -0,0 +1,36 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+namespace chip {
+
+/**
+ * @brief Create ZCL string from char
+ */
+static uint8_t * MakeZclCharString(uint8_t * zclString, const char * cString)
+{
+    size_t len = strlen(cString);
+    if (len > 254)
+    {
+        len = 0;
```

Comment:
I guess we could, but why would we not just clamp length to 254 instead?  Or more generally buffer size minus 1?

Reply:
The previous function with buffer size as a parameter cString[BufferSize] is now just cString *. Same as zclString, which was zclString[BufferSize] now is zclString *. 

It's due to making the ZclString function works for [this code](uint8_t * ToZclCharString(uint8_t * zclString, const char * cString, uint8_t maxLength)) here as well.

Reply:
I understand the change that was made, but what I am not sure about is why we did not change the signature to include the length of the buffer available for write, like the code linked to does....  Not passing in the buffer length here seems like it's just asking for buffer overruns.